scripts/gcp_caper_server/create_instance.sh

```diff
@@ -0,0 +1,314 @@
+ROOT_CAPER_CONF_DIR=/root/.caper
+GLOBAL_CAPER_CONF_FILE="$CAPER_CONF_DIR/default.conf"
+REMOTE_KEY_FILE="$CAPER_CONF_DIR/service_account_key.json"
+
+# prepend more init commands to the startup-script
+STARTUP_SCRIPT="""#!/bin/bash
+### make caper's work directory
+sudo mkdir -p $CAPER_CONF_DIR
+sudo chmod +r $CAPER_CONF_DIR
+
+### make caper's out/localization directory
+sudo mkdir -p $CAPER_CONF_DIR/local_loc_dir $CAPER_CONF_DIR/local_out_dir
+sudo chmod 777 -R $CAPER_CONF_DIR/local_loc_dir $CAPER_CONF_DIR/local_out_dir
+
+### make caper conf file
+cat <<EOF > $GLOBAL_CAPER_CONF_FILE
+# caper
+backend=gcp
+no-server-heartbeat=True
+# cromwell
+max-concurrent-workflows=300
+max-concurrent-tasks=1000
+# local backend
+local-out-dir=$CAPER_CONF_DIR/local_out_dir
+local-loc-dir=$CAPER_CONF_DIR/local_loc_dir
+# gcp backend
+gcp-prj=$GCP_PRJ
+gcp-region=$GCP_REGION
+gcp-out-dir=$GCP_OUT_DIR
+gcp-loc-dir=$GCP_LOC_DIR
+gcp-service-account-key-json=$REMOTE_KEY_FILE
+use-google-cloud-life-sciences=True
+# metadata DB
+db=postgresql
+postgresql-db-ip=$POSTGRESQL_DB_IP
+postgresql-db-port=$POSTGRESQL_DB_PORT
+postgresql-db-user=$POSTGRESQL_DB_USER
+postgresql-db-password=$POSTGRESQL_DB_PASSWORD
+postgresql-db-name=$POSTGRESQL_DB_NAME
+EOF
+sudo chmod +r $GLOBAL_CAPER_CONF_FILE
+
+### soft-link conf file for root
+sudo mkdir -p $ROOT_CAPER_CONF_DIR
+sudo ln -s $GLOBAL_CAPER_CONF_FILE $ROOT_CAPER_CONF_DIR
+
+### google auth shared for all users
+sudo touch $GCP_AUTH_SH
+echo \"gcloud auth activate-service-account --key-file=$REMOTE_KEY_FILE\" >> $GCP_AUTH_SH
+echo \"mkdir -p ~/.caper\" >> $GCP_AUTH_SH
+echo \"ln -s /opt/caper/default.conf ~/.caper/ 2> /dev/null | true\" >> $GCP_AUTH_SH
+
+$STARTUP_SCRIPT
+"""
+
+# append more init commands to the startup-script
+STARTUP_SCRIPT="""$STARTUP_SCRIPT
+### init PostgreSQL for Cromwell
+sudo -u postgres createuser root -s
+sudo createdb $POSTGRESQL_DB_NAME
+sudo psql -d $POSTGRESQL_DB_NAME -c \"create extension lo;\"
+sudo psql -d $POSTGRESQL_DB_NAME -c \"create role $POSTGRESQL_DB_USER with superuser login password '$POSTGRESQL_DB_PASSWORD'\"
+
+### upgrade pip and install caper croo
+sudo python3 -m pip install --upgrade pip
+sudo pip install caper croo
+"""
+
+echo "$(date): Google auth with service account key file."
+gcloud auth activate-service-account --key-file="$GCP_SERVICE_ACCOUNT_KEY_JSON_FILE"
+#export GOOGLE_APPLICATION_CREDENTIALS="$GCP_SERVICE_ACCOUNT_KEY_JSON_FILE"
+
+echo "$(date): Creating an instance..."
+gcloud --project "$GCP_PRJ" compute instances create \
+  "$INSTANCE_NAME" \
+  --boot-disk-size="$BOOT_DISK_SIZE" \
+  --boot-disk-type="$BOOT_DISK_TYPE" \
+  --machine-type="$MACHINE_TYPE" \
+  --zone="$ZONE" \
+  --image="$IMAGE" \
+  --image-project="$IMAGE_PROJECT" \
+  --metadata startup-script="$STARTUP_SCRIPT" \
+  $GCLOUD_EXTRA_PARAM
+echo "$(date): Created an instance successfully."
+
+echo "$(date): Waiting for 30 seconds for the instance to spin up..."
+sleep 30
```

Comment:
Would be better to actually use the instance status. 30s isn't really enough time to wait for a VM either. You can use something like this:
```suggestion
while [[ $(gcloud compute instances describe "${INSTANCE_NAME}" --zone "${ZONE}" --format="value(status)") -ne "RUNNING" ]]; do
    echo "$(date): Waiting for 30 seconds for the instance to spin up..."
    sleep 30
done
```

Reply:
Thanks, this looks so good. will apply to the script in another commit.
Added `--project` to specify project. (`gcloud --project "$GCP_PRJ" compute ...`)